Compiler warning when building library with g++ on Ubuntu

Hello! When building cpputest with g++ 9.3.0 on Ubuntu 20.04 and enabling -Wundef I get the following warning

```
warning:
../libs/cpputest/src/Platforms/Gcc/UtestPlatform.cpp:178:8: warning: "clang_major" is not defined, evaluates to 0 [-Wundef]
178 | #if !((clang_major == 3) && (clang_minor == 0))
```
I wonder if there is an` #ifdef __clang__` missing in UtestPlatform.cpp?

--- a/src/Platforms/Gcc/UtestPlatform.cpp
+++ b/src/Platforms/Gcc/UtestPlatform.cpp
@@ -175,8 +175,10 @@ static int PlatformSpecificSetJmpImplementation(void (*function) (void* data), v
  * MacOSX clang 3.0 doesn't seem to recognize longjmp and thus complains about _no_return_.
  * The later clang compilers complain when it isn't there. So only way is to check the clang compiler here :(
  */
-#if !((__clang_major__ == 3) && (__clang_minor__ == 0))
-_no_return_
+#ifdef __clang__
+ #if !((__clang_major__ == 3) && (__clang_minor__ == 0))
+ _no_return_
+ #endif
 #endif
 static void PlatformSpecificLongJmpImplementation()
 {

